templates/text-embeddings/README.md

```diff
@@ -1,100 +1,257 @@
+    chunk_overlap=0,
+)
 
-Examining the sample input file:
+def chunk_row(row, text_column_name):
+    """Main chunking logic."""
+    if not isinstance(row[text_column_name], str):
+        print(
+            "Found row with missing input text, "
+            f"skipping embeddings computation for this row: {row}"
+        )
+        return []
 
-```
->>> sample_input = ray.data.read_parquet("sample_input/sample-input.parquet")
+    length = num_words(row[text_column_name]) * 1.2
+    if length < 20 or length > 4000:
+        return []
+    chunks = splitter.split_text(row[text_column_name])
+
+    new_rows = [
+        {
+            "id": str(uuid.uuid4()),
+            **row,
+            text_column_name: chunk,
+        } for chunk in chunks
+        # Only keep rows with valid input text
+        if isinstance(chunk, str)
+    ]
+    return new_rows
 
->>> sample_input
-Dataset(num_blocks=768, num_rows=10, schema={content: string})
 ```
 
-Examining the sample (expected) output file:
+We use Ray Data's `flat_map()` method to apply the chunking function we defined above to each row of the input dataset. We use `flat_map()` as opposed to `map()` because the `chunk_row()` function may return more than one output row for each input row.
+
+*Note*: Because Ray Datasets are executed in a streaming fashion, running the cell below will not trigger execution because the dataset is not being consumed yet.
 
+
+```python
+chunked_ds = ds.flat_map(
+    chunk_row,
+    # Pass keyword arguments to the chunk_row function.
+    fn_kwargs={"text_column_name": "item"},
+)
 ```
->>> sample_output = ray.data.read_parquet("sample_output/sample-output.parquet")
-
->>> sample_output
-Dataset(
-   num_blocks=768,
-   num_rows=15,
-   schema={
-      id: string,
-      content: string,
-      url: string,
-      dump: string,
-      segment: string,
-      __index_level_0__: int64,
-      values: list<item: float>
-   }
+
+## Step 4: Compute Embeddings
+
+We define the `ComputeEmbeddings` class to compute embeddings using a pre-trained Hugging Face model. The full implementation
+is in `util/utils.py`.
+
+Next, apply batch inference for all input data with the Ray Data [`map_batches`](https://docs.ray.io/en/latest/data/api/doc/ray.data.Dataset.map_batches.html) method. Here, you can easily configure Ray Data to scale the number of model instances.
+
+
+```python
+from util.utils import ComputeEmbeddings
+
+embedded_ds = chunked_ds.map_batches(
+    ComputeEmbeddings,
+    # Total number of embedding model instances to use.
+    concurrency=NUM_MODEL_INSTANCES,
+    # Size of batches passed to embeddings actor. Set the batch size to as large possible
+    # without running out of memory. If you encounter out-of-memory errors, decreasing
+    # batch_size may help. 
+    batch_size=1, 
+    fn_constructor_kwargs={
+        "text_column_name": "item",
+        "model_name": HF_MODEL_NAME,
+        "device": "cpu",
+        "chunk_size": CHUNK_SIZE,
+    },
```

Comment:
This is the main update, where we import `ComputeEmbeddings` class from `util/utils.py`. This allows us to maintain a public docs page while keeping the embeddings logic code hidden